pkg/render/egressgateway/egressgateway.go

```diff
@@ -448,11 +452,19 @@ func (c *component) getHealthTimeoutDs() string {
 	return ""
 }
 
-func (c *component) egwSecurityContextConstraints() *ocsv1.SecurityContextConstraints {
-	namespacedName := fmt.Sprintf("%s-%s", c.config.EgressGW.Namespace, c.config.EgressGW.Name)
+func (c *component) getSecurityContextConstraints() *ocsv1.SecurityContextConstraints {
+	scc := SecurityContextConstraints()
+	sort.Strings(c.config.NamespacedNames)
+	for _, egwNames := range c.config.NamespacedNames {
+		scc.Users = append(scc.Users, fmt.Sprintf("system:serviceaccount:%s", egwNames))
+	}
+	return scc
+}
+
+func SecurityContextConstraints() *ocsv1.SecurityContextConstraints {
```

Comment:
I feel like maybe I didn't explain myself very clearly in the last comment. Here is what I was imagining this code would look like:

```go
func (c *component) securityContextConstraints() *ocsv1.SecurityContextConstraints {
        // Create a slice of all the EGW service accounts on the cluster that we need to apply this SCC to.
        serviceAccountNames := []string
        for _, egw := range c.Config.NamespacedNames {
            serviceAccountNames = append(serviceAccountNames, fmt.Sprintf("system:serviceaccount:%s:%s", egw.Namespace, egw.Name)
        }
        sort.Strings(serviceAccountNames)

	return &ocsv1.SecurityContextConstraints{
		TypeMeta:                 metav1.TypeMeta{Kind: "SecurityContextConstraints", APIVersion: "security.openshift.io/v1"},
		ObjectMeta:               metav1.ObjectMeta{Name: namespacedName},
		ObjectMeta:               metav1.ObjectMeta{Name: OpenShiftSCCName},
		AllowHostDirVolumePlugin: true,
		AllowHostIPC:             false,
		AllowHostNetwork:     false,
		AllowHostPID:             false,
		AllowHostPorts:           false,
		AllowPrivilegeEscalation: ptr.BoolToPtr(true),
		AllowPrivilegedContainer: true,
		FSGroup:                  ocsv1.FSGroupStrategyOptions{Type: ocsv1.FSGroupStrategyRunAsAny},
		RunAsUser:                ocsv1.RunAsUserStrategyOptions{Type: ocsv1.RunAsUserStrategyRunAsAny},
		ReadOnlyRootFilesystem:   false,
		SELinuxContext:           ocsv1.SELinuxContextStrategyOptions{Type: ocsv1.SELinuxStrategyMustRunAs},
		SupplementalGroups:       ocsv1.SupplementalGroupsStrategyOptions{Type: ocsv1.SupplementalGroupsStrategyRunAsAny},
		Users: serviceAccountNames, 
		Volumes: []ocsv1.FSType{"*"},
		Users:                    []string{},
		Volumes:                  []ocsv1.FSType{"*"},
	}
}
```

The intention is to remove the logic from the controller that is inserting / removing entries from the list, and instead do this logic declaratively - given the same input, we always get the same output. It doesn't rely on us following a series of steps.

Reply:
> The reason being, when a EGW resource gets deleted, SCC must be updated and having egwNames outside the config can be helpful right? WDYT?

Ah, right. If one is deleted, we won't call through to `Render` will we?

This is unfortunately proving more tricky than I thought!

Reply:
Perhaps the right thing is for the `SecurityContextConstraint` to be a completely separate render component than the rest of the EGW resources. 

That way, we can render the SCC every time the Reconcile function is called.

Reply:
Yeah. Even if one is deleted, we wont call through to ```Render```. I will move PSP and SCC to a new component which takes care of global resources in a separate PR.